x/staking/keeper/liquid_stake.go

```diff
@@ -0,0 +1,395 @@
+
+	return authorizations
+}
+
+// Inserts the address into a queue where it will sit for 1 unbonding period
+// before the tokenize share lock is removed
+// Returns the completion time
+func (k Keeper) QueueTokenizeSharesAuthorization(ctx sdk.Context, address sdk.AccAddress) time.Time {
+	params := k.GetParams(ctx)
+	completionTime := ctx.BlockTime().Add(params.UnbondingTime)
+
+	// Append the address to the list of addresses that also unlock at this time
+	authorizations := k.GetPendingTokenizeShareAuthorizations(ctx, completionTime)
+	authorizations.Addresses = append(authorizations.Addresses, address.String())
+
+	k.SetPendingTokenizeShareAuthorizations(ctx, completionTime, authorizations)
+	k.SetTokenizeSharesUnlockTime(ctx, address, completionTime)
+
+	return completionTime
+}
+
+// Cancels a pending tokenize share authorization by removing the lock from the queue
+func (k Keeper) CancelTokenizeShareLockExpiration(ctx sdk.Context, address sdk.AccAddress, completionTime time.Time) {
+	authorizations := k.GetPendingTokenizeShareAuthorizations(ctx, completionTime)
+
+	updatedAddresses := []string{}
+	for _, expiringAddress := range authorizations.Addresses {
+		if address.String() != expiringAddress {
+			updatedAddresses = append(updatedAddresses, expiringAddress)
+		}
+	}
+
+	authorizations.Addresses = updatedAddresses
+	k.SetPendingTokenizeShareAuthorizations(ctx, completionTime, authorizations)
+}
+
+// Unlocks all queued tokenize share authorizations that have matured
+// (i.e. have waited the full unbonding period)
+func (k Keeper) RemoveExpiredTokenizeShareLocks(ctx sdk.Context, blockTime time.Time) (unlockedAddresses []string) {
+	store := ctx.KVStore(k.storeKey)
+
+	// iterators all time slices from time 0 until the current block time
+	prefixEnd := sdk.InclusiveEndBytes(types.GetTokenizeShareAuthorizationTimeKey(blockTime))
+	iterator := store.Iterator(types.TokenizeSharesUnlockQueuePrefix, prefixEnd)
+	defer iterator.Close()
+
+	unlockedAddresses = []string{}
+	for ; iterator.Valid(); iterator.Next() {
+		authorizations := types.PendingTokenizeShareAuthorizations{}
+		k.cdc.MustUnmarshal(iterator.Value(), &authorizations)
+
+		for _, addressString := range authorizations.Addresses {
+			k.RemoveTokenizeSharesLock(ctx, sdk.MustAccAddressFromBech32(addressString))
+			unlockedAddresses = append(unlockedAddresses, addressString)
+		}
+		store.Delete(iterator.Key())
+	}
+
+	return unlockedAddresses
+}
+
+// Calculates and sets the global liquid staked tokens and liquid shares by validator
+// The totals are determined by looping each delegation record and summing the stake
+// if the delegator has a 32-length address. Checking for a 32-length address will capture
+// ICA accounts, as well as tokenized delegations which are owned by module accounts
+// under the hood
+// This function must be called in the upgrade handler which onboards LSM, as
+// well as any time the liquid staking cap is re-enabled
```

Comment:
> This function must be called any time the liquid staking cap is re-enabled

How will this be done?

Reply:
This is a stale comment, [removed](https://github.com/cosmos/cosmos-sdk/pull/16747/commits/73e72b9ef30f2315adeeefa33aa2526cf23ba8b7)